src/dotty/tools/dotc/Resident.scala

```diff
@@ -23,7 +23,7 @@ import scala.annotation.tailrec
  */
 object Resident extends Driver {
 
-  object residentCompiler extends Compiler
+  @unshared object residentCompiler extends Compiler
```

Comment:
Why is resident compiler an object?
Couldn't you have several resident compilers in the same process?

I'd better have resident compiler be a class, which is instantiated by some other object that has a `main` method.
This would also remove the need for `@unshared` annotation. I believe if something is indeed unshared - it could be a class.